resources/home/dnanexus/run_workflows/utils/demultiplexing.py

```
@@ -0,0 +1,323 @@
+        job_url = (
+            f"platform.dnanexus.com/projects/"
+            f"{demultiplex_project.replace('project-', '')}"
+            "/monitor/job/"
+            f"{job.id}"
+        )
+
+        Slack().send(
+            f"Demultiplexing job failed!\n\nError: {err}\n\n"
+            f"Demultiplexing job: {job_url}"
+        )
+        raise dx.exceptions.DXJobFailureError()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)